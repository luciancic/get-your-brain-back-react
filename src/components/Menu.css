/* On phone */
#menu { 
    display: grid;
    grid-template-rows: 2fr 4fr;
    height: 100vh;
}
#menu-text {
    align-self: center;
}

#menu-text h1 { 
    font-size: 2rem; 
    padding: 0 1rem;
}
#menu-text p { display: none}
#menu-buttons { 
    display: grid;
    grid-template-rows: 1fr 1fr 1fr;
    place-items: center center;
    height: 80%;
}


/* On desktop */
@media screen and (min-width: 768px) {
    #menu { 
        grid-template-rows: none;
        grid-template-columns: 3fr 2fr;
        height: 100vh;
    }
    #menu-text { 
        display: grid; 
        grid-template-rows: 1fr 2fr
    }
    #menu-text h1 { 
        align-self: end; 
        padding: 0 1rem
    }
    #menu-text p { 
        display: block; 
        padding: 0 5rem; 
        align-self: start; 
        text-align: left
    }
    #menu-buttons { 
        display: grid; 
        grid-template-rows: 1fr 1fr 1fr; 
        place-items: center left; 
        padding: 10% ; 
        height: 100%
    }
}