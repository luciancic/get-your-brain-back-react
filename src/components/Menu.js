import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startGame} from '../actions/gameActions';
import './Menu.css';

class Menu extends Component {
    render() {
        return <div id='menu' className='center'>
            <section id='menu-text'>
                <h1>Get Your Brain Back!</h1>
                <p>Challenge your brain and improve your working memory. Inspired by <a href='https://wikipedia.org/wiki/N-back' target='_blank' rel="noopener noreferrer">n-back</a>.</p>
            </section>
            <section id='menu-buttons'>
                <Link to='/game'><button className='btn green'  onClick={this.props.startGame}>Play Game</button></Link>
                <Link to='/tutorial'><button className='btn blue' onClick={this.props.startGame}>Play Tutorial</button></Link>
                <Link to='/settings'><button className='btn orange'>Settings</button></Link>
            </section>
        </div>
    }
}

export default connect(null, { startGame })(Menu);