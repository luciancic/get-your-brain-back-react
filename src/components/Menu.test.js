import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from './Menu';

it('renders Menu', () => {
  expect( shallow( <Menu /> )).toMatchSnapshot();
});
