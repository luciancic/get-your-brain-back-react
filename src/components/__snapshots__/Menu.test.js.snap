// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Menu 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          id="menu-text"
        >
          <h1>
            Get Your Brain Back!
          </h1>
          <p>
            Challenge your brain and improve your working memory. Inspired by 
            <a
              href="https://wikipedia.org/wiki/N-back"
              rel="noopener noreferrer"
              target="_blank"
            >
              n-back
            </a>
            .
          </p>
        </section>,
        <section
          id="menu-buttons"
        >
          <Link
            replace={false}
            to="/game"
          >
            <button
              className="btn green"
            >
              Play Game
            </button>
          </Link>
          <Link
            replace={false}
            to="/tutorial"
          >
            <button
              className="btn blue"
            >
              Play Tutorial
            </button>
          </Link>
          <Link
            replace={false}
            to="/settings"
          >
            <button
              className="btn orange"
            >
              Settings
            </button>
          </Link>
        </section>,
      ],
      "className": "center",
      "id": "menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Get Your Brain Back!
            </h1>,
            <p>
              Challenge your brain and improve your working memory. Inspired by 
              <a
                href="https://wikipedia.org/wiki/N-back"
                rel="noopener noreferrer"
                target="_blank"
              >
                n-back
              </a>
              .
            </p>,
          ],
          "id": "menu-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Get Your Brain Back!",
            },
            "ref": null,
            "rendered": "Get Your Brain Back!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Challenge your brain and improve your working memory. Inspired by ",
                <a
                  href="https://wikipedia.org/wiki/N-back"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  n-back
                </a>,
                ".",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Challenge your brain and improve your working memory. Inspired by ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "n-back",
                  "href": "https://wikipedia.org/wiki/N-back",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "n-back",
                "type": "a",
              },
              ".",
            ],
            "type": "p",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/game"
            >
              <button
                className="btn green"
              >
                Play Game
              </button>
            </Link>,
            <Link
              replace={false}
              to="/tutorial"
            >
              <button
                className="btn blue"
              >
                Play Tutorial
              </button>
            </Link>,
            <Link
              replace={false}
              to="/settings"
            >
              <button
                className="btn orange"
              >
                Settings
              </button>
            </Link>,
          ],
          "id": "menu-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn green"
              >
                Play Game
              </button>,
              "replace": false,
              "to": "/game",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Play Game",
                "className": "btn green",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Play Game",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn blue"
              >
                Play Tutorial
              </button>,
              "replace": false,
              "to": "/tutorial",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Play Tutorial",
                "className": "btn blue",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Play Tutorial",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn orange"
              >
                Settings
              </button>,
              "replace": false,
              "to": "/settings",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Settings",
                "className": "btn orange",
              },
              "ref": null,
              "rendered": "Settings",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            id="menu-text"
          >
            <h1>
              Get Your Brain Back!
            </h1>
            <p>
              Challenge your brain and improve your working memory. Inspired by 
              <a
                href="https://wikipedia.org/wiki/N-back"
                rel="noopener noreferrer"
                target="_blank"
              >
                n-back
              </a>
              .
            </p>
          </section>,
          <section
            id="menu-buttons"
          >
            <Link
              replace={false}
              to="/game"
            >
              <button
                className="btn green"
              >
                Play Game
              </button>
            </Link>
            <Link
              replace={false}
              to="/tutorial"
            >
              <button
                className="btn blue"
              >
                Play Tutorial
              </button>
            </Link>
            <Link
              replace={false}
              to="/settings"
            >
              <button
                className="btn orange"
              >
                Settings
              </button>
            </Link>
          </section>,
        ],
        "className": "center",
        "id": "menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Get Your Brain Back!
              </h1>,
              <p>
                Challenge your brain and improve your working memory. Inspired by 
                <a
                  href="https://wikipedia.org/wiki/N-back"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  n-back
                </a>
                .
              </p>,
            ],
            "id": "menu-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get Your Brain Back!",
              },
              "ref": null,
              "rendered": "Get Your Brain Back!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Challenge your brain and improve your working memory. Inspired by ",
                  <a
                    href="https://wikipedia.org/wiki/N-back"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    n-back
                  </a>,
                  ".",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Challenge your brain and improve your working memory. Inspired by ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "n-back",
                    "href": "https://wikipedia.org/wiki/N-back",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "n-back",
                  "type": "a",
                },
                ".",
              ],
              "type": "p",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/game"
              >
                <button
                  className="btn green"
                >
                  Play Game
                </button>
              </Link>,
              <Link
                replace={false}
                to="/tutorial"
              >
                <button
                  className="btn blue"
                >
                  Play Tutorial
                </button>
              </Link>,
              <Link
                replace={false}
                to="/settings"
              >
                <button
                  className="btn orange"
                >
                  Settings
                </button>
              </Link>,
            ],
            "id": "menu-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn green"
                >
                  Play Game
                </button>,
                "replace": false,
                "to": "/game",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Play Game",
                  "className": "btn green",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Play Game",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn blue"
                >
                  Play Tutorial
                </button>,
                "replace": false,
                "to": "/tutorial",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Play Tutorial",
                  "className": "btn blue",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Play Tutorial",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn orange"
                >
                  Settings
                </button>,
                "replace": false,
                "to": "/settings",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Settings",
                  "className": "btn orange",
                },
                "ref": null,
                "rendered": "Settings",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
